### PR TITLE
Rule builder styling fixes.

### PR DESCRIPTION
**Pull request by jmchilton**
See commits for details - mix of BS4 fallout fixes and enhancements (including some requested by @martenson in #5822).

### CODE REVIEW

File: client/galaxy/scripts/components/RuleCollectionBuilder.vue [hunk 1]
```diff
@@ -338,7 +338,7 @@
                 <button @click="resetRulesAndState" :title="titleReset" class="creator-reset-btn btn rule-btn-reset">
                     {{ l("Reset") }}
                 </button>
-                <button @click="createCollection" :title="titleFinish" class="create-collection btn btn-primary rule-btn-okay" v-bind:class="{ disabled: !validInput }">
+                <button @click="createCollection" :title="titleFinish" class="create-collection btn btn-secondary btn-primary" v-bind:class="{ disabled: !validInput }">
```
Comment (martenson) [inline client/galaxy/scripts/components/RuleCollectionBuilder.vue]: having both btn-primary and btn-secondary classes on a button is superfluous I think

Comment (martenson) [reply client/galaxy/scripts/components/RuleCollectionBuilder.vue]: disregard, was amended in subsequent commit

**Comment by martenson** [comment]
I am seeing really strange behavior when trying to define columns. The dropdown menu will intermittently *not* create a new row after click even though it goes through the 'active' state and closes. Sometimes it takes 4 attempts, other times it works on first, and even other times it just won't create the row ever (there is only so many times I can click). There is no js error.


screenshot 2018-04-18 11 47 02

**Comment by jmchilton** [comment]
@martenson Not just you - I see it too. Hmm....

**Comment by jmchilton** [comment]
That problem isn't this branch I don't think, it is dev.

**Comment by jmchilton** [comment]
Looks like the fix is just to place the click handler on the outer li instead of the inner a target. Probably some spacing difference in Bootstrap 4 - makes sense that the tests wouldn't detect it since they literally click the a target.

**Comment by martenson** [comment]
yeah bs4 wants to have `<div><a>...` and not `<ul><li><a>...` but this is pretty horrible behavior, it looks like working but fails 'randomly'.

**Comment by jmchilton** [comment]
Oh that probably explains the random really bright green in the downdown that is quite jarring also. I'll switch to use divs and a.

**Comment by martenson** [comment]
@jmchilton no it does not, that is an actual change I made to increase contrast which is pretty bad by default in bs4. The color choice is weird on purpose, so I get feedback. ;)

**Comment by martenson** [comment]
I opened a PR against your branch: https://github.com/jmchilton/galaxy/pull/66 with some UX improvements (I hope).

The code has various `help` and `title` elements but often they do not seem to be reachable from the UI. I think we could improve accessibility of this component a lot if we consistently show titles on most/all of the active elements. For some critical places I would consider using a bs tooltip or even a popover.

MDN entry on `<label>` [accessibility](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#Accessibility_concerns).

**Comment by martenson** [comment]
In reoriented view the rules overflow into the input.

screenshot 2018-04-19 12 28 24

**Comment by jmchilton** [comment]
Merged your PR into #5940 and disabled the reorient mode - I don't think that was working out. I actually use the wider view of the rules in #5926 when there is not preview available but I think if there is tabular data that mode is always just awkward.

I feel like all of the title and help elements I've added so far do show up in the GUI as mouseovers titles - but they could be more widespread and more prominent. The tooltips in the library GUI do look nicer than the browser mouseovers here. I'll see what what I can do.

**Comment by jmchilton** [comment]
@martenson I've used bootstrap-vue to add many more tooltips and such to the rule builder in https://github.com/galaxyproject/galaxy/pull/5940, it made it very easy.

**Comment by martenson** [comment]
@jmchilton marvelous!

**Pull request merged**